src/app/app.component.ts

```diff
@@ -51,11 +52,21 @@ export class AppComponent implements OnInit {
     if (params.get('hideGoogle')) {
       this.globalVars.hideGoogle = true;
     }
-    
+
+    // Callback should only be used in mobile applications, where payload is passed through URL parameters.
+    if (params.get('callback')) {
+      try {
+        this.globalVars.callback = new URL(params.get('callback') as string);
```

Comment:
How is callback related to hostname? Do we want to enforce that the callback has the same hostname as the thing that called this? I don't think it's necessary but put some juices on it and lmk.